// @mongosh/shell-api/lib/deep-inspect/custom-inspect.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addCustomInspect = addCustomInspect;
const customInspectSymbol = Symbol.for('nodejs.util.inspect.custom');
function customDocumentInspect(depth, inspectOptions, inspect) {
    const newInspectOptions = {
        ...inspectOptions,
        depth: Infinity,
        maxArrayLength: Infinity,
        maxStringLength: Infinity,
    };
    const copyToInspect = Array.isArray(this) ? this.slice() : { ...this };
    delete copyToInspect[customInspectSymbol];
    return inspect(copyToInspect, newInspectOptions);
}
function addInspectSymbol(obj) {
    if (!obj[customInspectSymbol]) {
        try {
            Object.defineProperty(obj, customInspectSymbol, {
                value: customDocumentInspect,
                enumerable: false,
                writable: true,
                configurable: true,
            });
        }
        catch (_a) {
        }
    }
}
function addCustomInspect(obj) {
    if (Array.isArray(obj)) {
        addInspectSymbol(obj);
        for (const item of obj) {
            addCustomInspect(item);
        }
    }
    else if (obj &&
        typeof obj === 'object' &&
        obj !== null &&
        !obj._bsontype &&
        !['[object Date]', '[object RegExp]'].includes(Object.prototype.toString.call(obj))) {
        addInspectSymbol(obj);
        for (const value of Object.values(obj)) {
            addCustomInspect(value);
        }
    }
}
